doc/src/guides/contributing/dependencies.md

```diff
@@ -0,0 +1,222 @@
+faster than the built-in Python `int`. When gmpy2 is installed, it is used
+automatically by certain core functions that operate on integers, such as the
+[polys](polys-docs). See {ref}`polys-domainsref` for more details.
+
+### Interactive Use
+
+#### IPython
+
+#### Jupyter Notebook
+
+### LaTeX Parsing
+
+#### antlr-python-runtime
+
+#### python-clang
+
+### Logic
+
+#### pycosat
+
+#### pysat
+
+### Plotting
+
+#### matplotlib
+
+#### pyglet
+
+### lambdify
+
+#### numpy
+
+#### scipy
+
+#### cupy
+
+#### tensorflow
+
+#### numexpr
+
+### Code Generation
+
+#### NumPy
+
+#### Cython
+
+#### Aesara
+
+#### pymc3
+
+#### llvmlite
+
+#### Wurlitzer
+
+[Wurlitzer](https://github.com/minrk/wurlitzer) is a Python package that
+allows capturing output from C extensions. It is used by some of the tests in
+the `sympy.codegen` submodule. It is only used by the test suite. It is not
+used by any end-user functionality. If it is not installed, some tests will be
+skipped.
+
+### Optional SymEngine Backend
+
+#### python-symengine
+
+[SymEngine](https://symengine.org/) is a fast symbolic manipulation library,
+written in C++. The SymEngine Python bindings may be used as an optional
+backend for SymPy core. To do this, first install the SymEngine Python
+bindings (with `pip install symengine` or `conda install -c conda-forge
+python-symengine`) and run SymPy with the `USE_SYMENGINE=1` environment variable.
+
+Presently, the SymEngine backend is only used by the
+[sympy.physics.mechanics](classical_mechanics) and
+[sympy.liealgebras](lie-algebras) modules, although you can also interface
+with SymPy's SymEngine backend directly by importing things from
+`sympy.core.backend`:
+
+```
+>>> from sympy.core.backend import Symbol
+>>> # This will create a SymEngine Symbol object if the USE_SYMENGINE
+>>> # environment variable is configured. Otherwise it will be an ordinary
+>>> # SymPy Symbol object.
+>>> x = Symbol('x')
+```
+
+SymEngine backend support is still experimental, so certain SymPy functions
+may not work correctly when it is enabled.
+
+### Experimental Rubi Integrator
+
+#### MatchPy
+
+[MatchPy](https://matchpy.readthedocs.io/en/latest/) is a library for doing
+pattern matching. It is used in the experimental sympy.integrals.rubi module,
+but presently, it is not used anywhere else in SymPy. SymPy and MatchPy are
+able to interface with each other.
+
+
+
+### Other
+
+#### cloudpickle
+
+The [cloudpickle](https://github.com/cloudpipe/cloudpickle) package can be
+used to more effectively pickle SymPy objects than the built-in Python [pickle](https://docs.python.org/3/library/pickle.html).
+Some tests in `sympy.utilities.tests.test_pickling.py` depend on cloudpickle
+to run. It is not otherwise required for any SymPy function.
+
+#### Sage
+
+
+## Development Dependencies
+
+Typical development on SymPy does not require any additional dependencies
+beyond Python and mpmath.
```

Comment:
pytest? (Not sure though, I haven't really understood the relation fully...)

Reply:
Added something on pytest. Let me know if that clears it up or if I should add more.

Reply:
We should add something about pytest somewhere with more detail. Personally I only use pytest for running the sympy tests.